L2/hardfork_devnet_test.md

```diff
@@ -0,0 +1,91 @@
+    - `l2GenesisBlobTimeOffset`
+### Validation after deployment:
+  - **SGT Validation**: Executing "Spend SGT without native gas token" should result in:
+    ```bash
+    Error: Failed to estimate gas: server returned an error response: error code -32000: gas required exceeds allowance (0)
+    ```
+  - **L2Blob Validation**: Transactions involving L2Blob should yield:
+    ```bash
+    Error: server returned an error response: error code -32000: transaction type not supported
+    ```
+## 2. Updating the Superchain Registry
+Perform the following steps to update the superchain registry:
+  - Clone and checkout the devnet branch
+    ```bash
+    git clone https://github.com/QuarkChain/superchain-registry.git
+    cd superchain-registry
+    git checkout devnet
+    ```
+  - Update configuration files:
+      - Modify `./chainList.toml` and `./chainList.json` based on your deployment specifics.
+      - Edit `./superchain/config/sepolia/qkc.toml`
+          - Set `l2_blob_time` and `soul_gas_token_time` accordingly
+          - Update values in `genesis.l1`, `genesis.l2` and `genesis.system_config`
+          - Ensure all relevant roles and addresses are correctly set.
+  - Generate and compress genesis file:
+    - Download genesis.json into your working directory and rename it:
+      ```bash
+      mv genesis.json qkc.json
+      ```
+    - Install compression tool if necessary:
+      ```bash
+      apt install zstd
+      ```
+    - Compress genesis file using the provided dictionary:
+      ```bash
+      zstd -D ./superchain/extra/dictionary qkc.json
+      ```    
+    - Move compressed file to correct location:
+      ```bash
+      mv qkc.json.zst ./superchain/extra/genesis/sepolia/
+      ```        
+  - Commit and push changes:
+    ```bash
+    git add .
+    git commit -m "Update genesis and config for L2Blob and SGT enabling"
+    git push origin devnet    
+    ```  
+## 3. Generating the Latest `superchain-configs.zip`
+We will generate the zip file on the deployment machine directly.
+  - Modify `op-geth` source:
+    - Navigate to the `op-geth` folder.
+    - Update repository URL in `sync-superchain.sh`
+      ```bash
+      - https://github.com/ethereum-optimism/superchain-registry.git
+      + https://github.com/quarkchain/superchain-registry.git      
+      ```
+    - Update commit hash in `superchain-registry-commit.txt` to match the latest commit on the devnet branch.
+  - Rebuild binaries: 
+    - For op-geth: 
+      ```bash
+      make geth
+      ```
+    - For op-node: 
+      - Copy `superchain-configs.zip` to the appropriate library folder.
+      - Temporarily comment out the CheckL1ChainID validation [here](https://github.com/QuarkChain/optimism/blob/06a9487cb7f3b9398de0b9ba27896e7a4ef9d1c0/op-node/rollup/types.go#L195)
+      - Compile op-node:
+        ```bash
+        cd ../optimism
+        just op-node/op-node
+        ```
+## 4. Restart op-node && op-geth
+  - Restart op-node
```

Comment:
Maybe for mainnet, we should dump& compare the result of `LoadOPStackGenesis`/`LoadOPStackRollupConfig` before actually apply it.

Reply:
For a mainnet hardfork release, we should first commit the updated `superchain-configs.zip` and `superchain-registry-commit.txt` to the op-geth repository. Afterward, we’ll build the final release binary and validate it thoroughly in our staging environment to ensure everything operates as expected. Once testing is complete and confirmed successful, we can then officially release the final version and deploy to production.

Reply:
Yeah, but before actual testing, we can just compare the json result of genesis and rollup config to ensure no field is modified unexpectedly, this can save much time.

Reply:
Added a note [here](https://github.com/QuarkChain/pm/pull/64/commits/c08c96b8e307e1fa99965e263ea373abef69c787)